### ISSUE TITLE
Deployment Issue in Step 6 - Updating Constants.resx

### ISSUE THREAD

**Issue by laughingmanzero** [Microsoft/mattercenter#100]
I've tried finding the root of this problem for the last day, it seems to have something to do with the Microsoft.Legal.MatterCenter.ProviderService\App_GlobalResources\Constants.resx file.

```
3/9/2016 1:06:48 PM - Step 6: Update resource and config files in build
Updating Constants.resx file
Result Source ID not found.
Message: Result Source ID not found.
Stacktrace:    at Microsoft.Legal.MatterCenter.UpdateAppConfig.UpdateConfig.GetResultSourceId(String login, String password)
Message: Value cannot be null.
Parameter name: value
Stacktrace:    at System.Xml.Linq.XElement.set_Value(String value)
   at Microsoft.Legal.MatterCenter.UpdateAppConfig.UpdateConfig.UpdateResourceConstants(Dictionary`2 updateList, String constantElement, String constantAttribute, List`1 manifestFile)
   at Microsoft.Legal.MatterCenter.UpdateAppConfig.UpdateConfig.UpdateConstants(Dictionary`2 updateList, String folderName)
Message: Failed to update app schema files
Failed to update resource files
Updated Web.config
3/9/2016 1:06:50 PM - Updating resource and config files in solution failed
```

It is referenced in the deployment guide that I should not change the value unless performing manual install here: Search_Result_Source_ID (Only if doing manual deployment, otherwise scripts will handle)

Here's what I have in the Microsoft.Legal.MatterCenter.ProviderService\App_GlobalResources\Constants.resx file:

```
<data name="Search_Result_Source_ID" xml:space="preserve">
    <value>[Enter Result source ID which retrieves results from Matter Center specific site collection only, e.g. 8413cd39-2156-4e00-b54d-11efd9abdb89</value>
    <comment>Result source ID which retrieves results from Matter Center specific site collection only</comment>
  </data>
```

**Comment by DaveyDoo** [comment]
Are you sure the field Search_Result_Source_ID isn't in the ProviderService/App_GlobalResources/Constants.resx file?  I would try inserting the GUID indicated.

**Comment by laughingmanzero** [comment]
Sorry, in the above example I had copy/pasted the wrong directory. I'll update the first post to reflect the ProviderService file location. I must have pasted in that sharepoint one without enough coffee earlier lol.

When you say the GUID indicated, do you mean the one included in the example? "8413cd39-2156-4e00-b54d-11efd9abdb89"?

**Comment by DaveyDoo** [comment]
Definitely not the GUID from the example.  It will either be a GUID generated by SharePoint online (sorry I don't remember which GUID that is) but there are also a couple of GUIDs you generate on your own.

**Comment by ADOrigin** [comment]
I'm assuming you've got the correct URL's and other values in the Config tab of the MCDeploymentConfig.xlsx file.

I'm not sure if this will help you, since my issue probably wasn't exactly the same as yours, but I found that even after deploying "successfully", this GUID was still not set in this constants.resx file.  What I ended up doing was decompiling Microsoft.Legal.MatterCenter.UpdateAppConfig.exe (in the /deployments/scripts/Helper Utilities folder).  I used Telerik JustDecompile because it's free and works pretty well.  

There's a method in there called GetResultSourceID and another one called GetResultSource, so I copied that code out and pasted it into a new Visual Studio console .exe project so that I could debug it.  I ran that (in the call to ConfigureClientContext, substitute in the address of your SharePoint site, your username and password as the first 3 parameters).  At the end of the call to UpdateConfig.GetResultSource, add a watch on the "element" variable and look at it with the XML visualizer.  You'll see something like...

```
<d4p1:Source ...>
  ....
  <d4p1:Id>[GUID]</d4p1:Id>
  ...
  <d4p1:Name>Matter Center</d4p1:Name>
  ...
</d4p1:Source>

```
That value in [GUID] is what I manually put into the constants.resx file.  I'm still working on getting this thing up and running, so I'm not yet 100% sure that it's the correct GUID, but I'm reasonably confident based on what I saw in the decompiled UpdateAppConfig.exe code.  I'm definitely not getting the error message I was previously getting on the search page.
